libkodimote/mpris2/mprisplayer.cpp

```diff
@@ -0,0 +1,420 @@
+    }
+
+    QTime time = QTime(0, 0, 0).addMSecs(newTime);
+    //TODO: use m_player->totalTime from player improvements branch
+    if (time > m_player->currentItem()->duration()) {
+        m_player->skipNext();
+    } else {
+        m_player->seek(time);
+    }
+}
+
+void MprisPlayer::SetPosition(QDBusObjectPath path, qint64 position)
+{
+    if (!m_player || !m_player->currentItem()) {
+        return;
+    }
+
+    if (buildPath(m_player->currentItem()) != path.path()) {
+        return;
+    }
+
+    QTime time = QTime(0, 0, 0).addMSecs(position / 1000);
+    m_player->seek(time);
+}
+
+void MprisPlayer::OpenUri(QString uri)
+{
+    m_protocols->execute(uri);
+}
+
+QString MprisPlayer::buildPath(LibraryItem *item) const
+{
+    if (item->episodeId() != -1) {
+        return QString("/org/mpris/MediaPlayer2/Track/episode/%1").arg(item->episodeId());
+    } else if (item->movieId() != -1) {
+        return QString("/org/mpris/MediaPlayer2/Track/movie/%1").arg(item->movieId());
+    } else if (item->musicvideoId() != -1) {
+        return QString("/org/mpris/MediaPlayer2/Track/musicVideo/%1").arg(item->musicvideoId());
+    } else if (item->channelId() != -1) {
+        return QString("/org/mpris/MediaPlayer2/Track/channel/%1").arg(item->channelId());
+    } else if (item->songId() != -1) {
+        return QString("/org/mpris/MediaPlayer2/Track/song/%1").arg(item->songId());
+    } else if (item->recordingId() != -1) {
+        return QString("/org/mpris/MediaPlayer2/Track/recording/%1").arg(item->recordingId());
+    }
+
+    return "";
+}
+
+void MprisPlayer::sendPropertyChanged(const QString &property)
+{
+    sendPropertiesChanged(QStringList(property));
+}
+
+void MprisPlayer::sendPropertiesChanged(const QStringList &properties)
+{
+    QDBusMessage signal = QDBusMessage::createSignal("/org/mpris/MediaPlayer2", "org.freedesktop.DBus.Properties", "PropertiesChanged");
+    signal << "org.mpris.MediaPlayer2.Player";
+    QVariantMap changedProps;
+    foreach (QString property, properties) {
+        changedProps.insert(property, this->property(property.toUtf8()));
+    }
+    signal << changedProps;
+    signal << QStringList();
+    QDBusConnection::sessionBus().send(signal);
+}
+
+void MprisPlayer::activePlayerChanged()
+{
+    if (m_player == Kodi::instance()->activePlayer()) {
+        return;
+    }
+
+    if (m_player) {
+        disconnect(m_player, SIGNAL(stateChanged()), this, SLOT(stateChanged()));
+        disconnect(m_player, SIGNAL(currentItemChanged()), this, SLOT(currentItemChanged()));
+        disconnect(m_player, SIGNAL(timeChanged()), this, SLOT(timeChanged()));
+        disconnect(m_player, SIGNAL(speedChanged()), this, SLOT(speedChanged()));
+        disconnect(m_player, SIGNAL(shuffleChanged()), this, SLOT(shuffleChanged()));
+        disconnect(m_player, SIGNAL(repeatChanged()), this, SLOT(repeatChanged()));
+        disconnect(m_player->playlist(), SIGNAL(countChanged()), this, SLOT(playlistChanged()));
```

Comment:
I think you can just use this one: disconnect(myObject, 0, myReceiver, 0); like:
disconnect(m_player, 0, this, 0); then you don't need to list them all.